src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -53,14 +53,25 @@ public PaymentMessageProcessor(
     public boolean processNextMessage() throws ServiceBusException, InterruptedException {
         IMessage message = messageReceiver.receive();
         if (message != null) {
-            MessageProcessingResult result = process(message);
-            tryFinaliseProcessedMessage(message, result);
+            if (message.getLabel() == null) {
+                deadLetterTheMessage(message, "Missing label", null);
+            } else {
+                switch (message.getLabel()) {
+                    case "CREATE":
+                        MessageProcessingResult result = processCreateCommand(message);
+                        tryFinaliseProcessedMessage(message, result);
+                        break;
+                    case "UPDATE":
+                    default:
+                        deadLetterTheMessage(message, "Unrecognised message type: " + message.getLabel(), null);
```

Comment:
perhaps best to avoid nulls. so shall we say reason is "unrecognised message type" and description is more elaborative, including label value?

Reply:
switch doesn't work with null

Reply:
i meant dead letter reason/description ;)